### PR TITLE
Upgrade bitcoin

### PR DESCRIPTION
**Pull request by tcharding**
Upgrade dependencies to use the latest `rust-bitcoin v0.31`.

While we are at it, bump the crate version ready for release.

(Includes pinning dependencies for MSRV build in CI.)

### CODE REVIEW

**Comment by tcharding** [comment]
Excuse the noise, this is another go at #78 because I saw #76.

This is #60 with pinning in CI. I previously handed off to @getong because he did #74 but that seems to be stalling. Lets see if I can get this past CI.

FTR I did the pinning with my nose i.e., by trial and error and searching crates on crates.io

**Comment by tcharding** [comment]
BOOM! Party time.

File: .github/workflows/cont_integration.yml [hunk 1]
```diff
@@ -51,10 +51,12 @@ jobs:
     - name: pin dependencies
       if: matrix.rust.version == '1.63.0'
       run: |
-        cargo update -p zstd-sys --precise "2.0.8+zstd.1.5.5"
+        cargo update -p zstd-sys --precise "2.0.6+zstd.1.5.2"
+        cargo update -p zstd --precise 0.11.2+zstd.1.5.2
         cargo update -p time --precise "0.3.20"
         cargo update -p jobserver --precise "0.1.26"
         cargo update -p home --precise 0.5.5
+        cargo update -p jobserver --precise 0.1.26
```
[the remark this answers is not in the document]
Comment (tcharding) [reply .github/workflows/cont_integration.yml:59]: Ouch, I thought I remove that locally. Will re-spin, thanks.

File: .github/workflows/cont_integration.yml [hunk 2]
```diff
@@ -51,10 +51,12 @@ jobs:
     - name: pin dependencies
       if: matrix.rust.version == '1.63.0'
       run: |
-        cargo update -p zstd-sys --precise "2.0.8+zstd.1.5.5"
+        cargo update -p zstd-sys --precise "2.0.6+zstd.1.5.2"
+        cargo update -p zstd --precise 0.11.2+zstd.1.5.2
```
[the remark this answers is not in the document]
Comment (tcharding) [reply .github/workflows/cont_integration.yml:55]: Oh, I remember. I brought `zstd-sys` back to be in line with the new `zst`, but like you say we may not need both.

**Comment by tcharding** [comment]
I left, `zstd` and `zstd-sys` in there, likely unnecessary but the additional explicitness doesn't hurt. Will remove if anyone really wants it gone.

**Comment by tcharding** [comment]
Thanks for the review @tnull!

**Comment by notmandatory** [comment]
I've merged #76 so this should be rebased on master so we just see e41b360876fa4f6dcc6bcffe5cf1d5ba716e942f.

**Comment by tcharding** [comment]
Done, thanks.

**Review by oleonardolima — COMMENTED** [review]
utACK 581cfb07cc8f48239ec915bb2a817ff25407572f

**Comment by notmandatory** [comment]
Overall looks good to me, just need to resolve above comments about zstd. 

@tnull is it OK with you to merge this one before #75? This one is a smaller change so should be easier to rebase on top of.

File: .github/workflows/cont_integration.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tcharding) [reply .github/workflows/cont_integration.yml]: Changed as requested, lets let CI prove if it works for us

**Comment by tcharding** [comment]
You guys were right about the pins, thanks for pushing me.

**Review by notmandatory — APPROVED** [review]
utACK c0460471236db834aafa0a4d618d90fc9b1d25ef

**Review by oleonardolima — APPROVED** [review]
ACK

**Pull request merged**